qcfractal/tests/test_collections.py

```diff
@@ -1368,3 +1369,13 @@ def test_get_collection_no_records_ds(fractal_compute_server):
     ds.get_entries()
     assert len(ds.data.records) == 1
     assert ds.data.records[0].name == "He1"
+
+
+def test_collection_metadata(fractal_compute_server):
+    client = ptl.FractalClient(fractal_compute_server)
+
+    ds = ptl.collections.Dataset("test_collection_metadata", client=client)
+    ds.data.metadata["data_points"] = 133_885
```

Comment:
This test indicates that `metadata` is not `Optional` which implies a `None` value is acceptable. Good example of how Pydantic is clarifying our thinking on `Optional`.

Reply:
It's a little tricky since existing collections will not have metadata. I wanted to allow for this possibility, but also to make new collections have an empty metadata dict going forward. Test has been updated accordingly.

Reply:
Lets do a data migration in this case. Add a new `metadata` column with `{}` as default. Should save us some headaches in the future.

Reply:
I thought we decided against this due to the partially migrated state of the collections table on molssi10

Reply:
Need to chat about this in person, but I think we need to hand alter the tables back to v0.11 schema anyways.